creator-node/test/CNodeHealthManager.test.js

```diff
@@ -0,0 +1,478 @@
+/* eslint-disable no-unused-expressions */
+const nock = require('nock')
+const chai = require('chai')
+const sinon = require('sinon')
+const { expect } = chai
+chai.use(require('sinon-chai'))
+chai.use(require('chai-as-promised'))
+const proxyquire = require('proxyquire')
+
+const CNodeHealthManager = require('../src/services/stateMachineManager/CNodeHealthManager')
+const config = require('../src/config')
+const Utils = require('../src/utils')
+
+describe('test CNodeHealthManager -- getUnhealthyPeers()', function () {
+  let sandbox
+  beforeEach(function () {
+    sandbox = sinon.createSandbox()
+  })
+
+  afterEach(function () {
+    sandbox.restore()
+  })
+
+  const healthyCn1 = 'http://healthy_cn1.co'
+  const healthyCn2 = 'http://healthy_cn2.co'
+  const unhealthyCn3 = 'http://unhealthy_cn3.co'
+  const unhealthyCn4 = 'http://unhealthy_cn4.co'
+  const healthyNodes = [healthyCn1, healthyCn2]
+  const unhealthyNodes = [unhealthyCn3, unhealthyCn4]
+  const users = [
+    {
+      user_id: 1,
+      wallet: 'wallet1',
+      primary: healthyCn1,
+      secondary1: unhealthyCn3,
+      secondary2: unhealthyCn3
+    },
+    {
+      user_id: 2,
+      wallet: 'wallet2',
+      primary: unhealthyCn4,
+      secondary1: healthyCn2,
+      secondary2: healthyCn1
+    }
+  ]
+  const thisContentNodeEndpoint = 'http://healthy_cn1.co'
+
+  it('returns all unhealthy nodes with default performSimpleCheck', async function () {
+    // Stub functions that getUnhealthyPeers() will call
+    sandbox
+      .stub(CNodeHealthManager, '_computeContentNodePeerSet')
+      .returns(new Set([...healthyNodes, ...unhealthyNodes]))
+    const isNodeHealthyStub = sandbox.stub(CNodeHealthManager, 'isNodeHealthy')
+    isNodeHealthyStub.withArgs(sinon.match.in(healthyNodes)).resolves(true)
+    isNodeHealthyStub.withArgs(sinon.match.in(unhealthyNodes)).resolves(false)
+
+    // Verify that the correct unhealthy peers are returned
+    return expect(
+      CNodeHealthManager.getUnhealthyPeers(users, thisContentNodeEndpoint)
+    ).to.eventually.be.fulfilled.and.deep.equal(new Set(unhealthyNodes))
+  })
+
+  for (const performSimpleCheck of [true, false]) {
```

Comment:
this is an interesting pattern. I wouldve written two explicit and static unit tests to test how the fn performed with `performSimpleCheck` = true or false. or, since the setup is the same, just done 2 expect checks with t/f.

not a big deal -- would prob prefer explicit and static tests in these scenarios since they're a bit easier to read

Reply:
hmm I guess I can do an extra expect. they seem like pretty equal solutions to me but I guess it doesn't hurt to be explicit

Reply:
although I'm not sure @vicky-g wdyt about [these tests](https://github.com/AudiusProject/audius-protocol/blob/master/identity-service/test/routes/trackListensTest.js#L103). I think they would've been a lot uglier if I had wrote them explicitly

Reply:
I personally wouldve preferred the tests to be broken up like
```
it('if performSimpleCheck is enabled, this should happen', function () {....})
it('if performSimpleCheck is disabled, this should happen', function () {....}) 
```

just to isolate distinct flows and distinct results, and if a test fails, we know it was either the true or false case that failed, vs having to run the specific unit test to check

and I think its okay to be more verbose in unit tests
verbose unit tests >>>> verbose code 😉

Reply:
I dont think there needs to be changes now. but for future reference I think its easier to read to make flows distinct!

Reply:
synced offline. I'll go ahead and break up the for loops 👍